File: File Reading Examples/Python API/Channel.py
from enum import Enum
from uuid import UUID
import numpy as np

class Channel:
    """Channel class wrapper for FileReaderAPI"""

    def __init__(self, channel, data):
        self.channel = channel
        self.data = data

    def Name(self) -> str:
        """Channel Name - This name can be set prior to a data stream"""

        name = str(self.channel.Name)
        return name

    def SampleRate(self) -> float:
        """Channel Sample Rate - The amount data points per second this channel was collected at"""

        sampleRate = float(self.channel.SampleRate)
        return sampleRate

    def Units(self) -> str:
        """Channel Unit - The unit of measurement for this channel"""

        units = str(Units(int(self.channel.Units)).name)
        return units

    def RangeMin(self) -> float:
        """Channel Minimum Range - The minimum value a data point can be"""

        rangeMin = float(self.channel.RangeMin)
        return rangeMin

    def RangeMax(self) -> float:
        """Channel Maximum Range - The maximum value a data point can be"""

        rangeMax = float(self.channel.RangeMax)
        return rangeMax

    def LogThisChannel(self) -> bool:
        """Channel Logging - Was this channel displayed during data collection"""

        logThisChannel = bool(self.channel.LogThisChannel)
        return logThisChannel

    def InternalName(self) -> str:
        """Channel Logging - Was this channel displayed during data collection"""

        internalName = str(self.channel.InternalName)
        return internalName

    def ChannelType(self) -> str:
        """Channel Type - Type of channel (ie. EMG, ACC, GYRO)"""

        type = str(self.channel.ChannelType)
        return type

    def SamplesPerFrame(self) -> int:
        """Channel Width - Amount of data points received on this channel per frame"""

        channelWidth = int(self.channel.SamplesPerFrame)
        return channelWidth


    def LocalChannelNumber(self):
        """Channel Local Index - Channel index based on all the sensor component channels"""

        localChannelNumber = int(self.channel.LocalChannelNumber)
        return localChannelNumber

    def Guid(self) -> str:
        return self.channel.GuidString;

    def Data(self):
        """Channel Data - All of the data associated with this channel during the collection"""
        return list(self.channel.GetYData(self.data))

class Units(Enum):
        Unknown = 0
        VOLTS = 1
        MILLIVOLTS = 2
        G = 3
        MICROTESLA = 4
        DEG_S = 5
        DEGS = 6
        Quaternion = 7
        QuaternionAccuracy = 8
        Orientation = 9
        OrientationAccuracy = 10
        Hz = 11
        Percentage = 12
        N_4cm2 = 13
        N = 14
        Revolutions = 15
        g_dl = 16
        BPM = 17
        METERS_SECOND = 18
        METERS = 19
        RPM = 20
        RADIANS_S = 21
        Nm = 22
        WATTS = 23
        Amplitude = 24
        Kilograms = 25
        Cycle = 26
        L = 27
        L_min = 28
        mL_kg_min = 29
        BrPM = 30,
        mL_min = 31

File: File Reading Examples/MATLAB API/Channel.m
classdef Channel < handle
    %Channel class wrapper for FileReaderAPI%   
    
    properties (Access = private)
       channel;
       datastream;
    end
    
    methods  
        function obj = Channel(channel, datastream)
        %Returns channel object%
        
           obj.channel = channel;
           obj.datastream = datastream;
        end
        
        function name = Name(obj)
        %Channel Name - This name can be set prior to a data stream%
        
            name = string(obj.channel.Name);
        end
        
        function sampleRate = SampleRate(obj)
        %Channel Sample Rate - The amount data points per second this
        %channel was collected at%
        
            sampleRate = double(obj.channel.SampleRate);
        end
        
        function units = Units(obj)
        %Channel Unit - The unit of measurement for this channel%
        
            units = string(obj.channel.Units);
        end
        
        function rangeMin = RangeMin(obj)
        %Channel Minimum Range - The minimum value a data point can be%
        
            rangeMin = double(obj.channel.RangeMin);
        end
        
        function rangeMax = RangeMax(obj)
        %Channel Maximum Range - The maximum value a data point can be%
        
            rangeMax = double(obj.channel.RangeMax);
        end
        
        function logChannel = LogThisChannel(obj)
        %Channel Logging - Was this channel displayed during data collection%
        
            logChannel = obj.channel.LogThisChannel;
        end
        
        function internalName = InternalName(obj)
        %Channel Logging - Was this channel displayed during data collection%
        
            internalName = string(obj.channel.InternalName);
        end
        
        function channelType = ChannelType(obj)
        %Channel Type - Type of channel (ie. EMG, ACC, GYRO)%
        
            channelType = string(obj.channel.ChannelType);
        end
        
        function samplesPerFrame = SamplesPerFrame(obj)
        %Channel Samples Per Frame - Amount of data points received on this channel per frame%
        
            samplesPerFrame = double(obj.channel.SamplesPerFrame);
        end
        
        
        function localChannelNumber = LocalChannelNumber(obj)
        %Channel Local Index - Channel index based on all the sensor component channels%
        
            localChannelNumber = double(obj.channel.LocalChannelNumber);
        end
        
        function guid = Guid(obj)
            
            guid = string(obj.channel.GuidString);
            
        end
        
        function parsedData = Data(obj)
        %Channel Data - All of the data associated with this channel during the collection%
            parsedData = double(obj.channel.GetYData(obj.datastream));
        end
    end
end

